File: samples/fizzbuzz.lua
function addOperation(x)
    return x + 2
end

function fizzBuzzOperation(num)
    local output = ""
    if num % 3 == 0 then output = output .. "Fizz" end
    if num % 5 == 0 then output = output .. "Buzz" end
    if output == "" then return tostring(num) else return output end
end

function processNumbers(numbers, operation)
    for _, num in ipairs(numbers) do
        local status, result = pcall(operation, num)
        if status then
            print(result)
        else
            print("Error:", result)
        end
    end
end

local numbers = {}
for i = 1, 15 do table.insert(numbers, i) end
processNumbers(numbers, fizzBuzzOperation)

File: samples/fizzbuzz.js
const addOperation = x => x + 2;
const fizzBuzzOperation = num => {
    let output = '';
    if (num % 3 === 0) output += 'Fizz';
    if (num % 5 === 0) output += 'Buzz';
    return output || num;
};

const processNumbers = async (numbers, operation) => {
    try {
        return numbers.map(operation);
    } catch (error) {
        console.error(`Error processing numbers: ${error}`);
    }
};

const numbers = Array.from({ length: 15 }, (_, i) => i + 1);

processNumbers(numbers, fizzBuzzOperation).then(console.log);